src/drivers/amd/renoir/ipc.c

```diff
@@ -120,7 +120,7 @@ static void irq_handler(void *arg)
 	}
 }
 
-enum task_state ipc_platform_do_cmd(void *data)
+enum task_state ipc_platform_do_cmd(struct ipc *ipc)
```

Comment:
since the sof->ipc is global so we actually don't need waste the stack memory in all these callings?

Reply:
I'm not sure passing a function parameter increases stack usage. AFAIK in the Xtensa function API the first couple of function parameters are passed in registers, as is also common on other architectures. I do agree that some global variables don't really have to be passed as parameters - we have `ipc_get()` and similar functions, but we already aren't perfectly consistent about our decisions on whether to use that function or pass a parameter. And TBH I'm not 100% certain what the best way should be. I actually think it's good to pass it as a parameter internally and only use `ipc_get()` where parameter passing is impossible... So, I think this is a bit more complex and OTOH not very important...

Reply:
> I'm not sure passing a function parameter increases stack usage. AFAIK in the Xtensa function API the first couple of function parameters are passed in registers, as is also common on other architectures. I do agree that some 

But the function calling stack could be quite deep so the total amount of parameters could be more then 'couple' and eventually increase the stack usage.

> global variables don't really have to be passed as parameters - we have `ipc_get()` and similar functions, but we already aren't perfectly consistent about our decisions on whether to use that function or pass a parameter. And TBH I'm not 100% certain what the best way should be. I actually think it's good to pass it as a parameter internally and only use `ipc_get()` where parameter passing is impossible... So, I think this is a bit more complex and OTOH not very important...

Yes it's minor thing actually so no blocker for my approval, I asked for this since seeing that you are in it -- changing of it :)